File: src/main/java/ar/edu/unq/epers/unidad5/model/Producto.java
package ar.edu.unq.epers.unidad5.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.ArrayList;
import java.util.List;

@ToString
@Setter
@Getter
@EqualsAndHashCode
@NoArgsConstructor(access = AccessLevel.PROTECTED)

@Document("Producto")
public class Producto {

    @Id
    private String id;
    private String codigo;
    private String nombre;
    private String marca;
    private List<Precio> precios = new ArrayList<>();

    public Producto(String codigo, String nombre, String marca) {
        this.codigo = codigo;
        this.nombre = nombre;
        this.marca = marca;
    }

    public void addPrecio(Precio precio) {
        this.precios.add(precio);
    }
}

File: src/main/kotlin/ar/edu/unq/epers/unidad5/model/Producto.kt
package ar.edu.unq.epers.unidad5.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document("Producto")
class Producto {
    @Id
    var id: String? = null
    lateinit var codigo: String
    var nombre: String? = null
    var marca: String? = null
    var precios: MutableList<Precio> = mutableListOf()

    //Se necesita un constructor vacio para que jackson pueda
    //convertir de JSON a este objeto.
    protected constructor() {
    }
    constructor(codigo: String, nombre: String, marca: String) {
        this.codigo = codigo
        this.nombre = nombre
        this.marca = marca
    }

    fun addPrecio(precio: Precio) {
        precios.add(precio)
    }

}